### DISCUSSION TITLE
Do I need to release collection after every load operation?

### DISCUSSION THREAD

**Discussion by StormstoutY** [milvus-io/milvus#27771] [Q&A and General discussion] [answered]
I load before each search or query operation. I wonder do I need to release the collection after each load operation? Or will milvus automatically release this after the current thread or connection is terminated?

In addition, I used the default helm chart to deploy the milvus cluster, but I noticed that the request/limit was not set for milvus-related pods in the chart (so these pods can use all available resources on the assigned k8s nodes?). If I want to improve the performance of milvus, what pod restrictions do I need to modify?

**Comment by yhmo** [comment, marked as answer]
After you call load(), the milvus server will read the collection's data into memory for search. It is loaded on the server-side, not on the client-side. So, if your client app exit, the collection will be kept in memory of the server. Connection termination doesn't affect the server-side behavior, either.
If the server shuts down, the memory is released.
Call release() once after you finish your search actions, the server will release the memory.

Typical workflow:  create_collection ----> create_index ----> insert ---> load ----> search ----> search ----> .... ----> search ---> release

Assign resources (cpu cores, memory capacity) to each pod. There is a tool to estimate the required resources for each pod: https://milvus.io/tools/sizing/   input your dataset size and choose index type, the tool recommends resources for each pod.

**Reply by StormstoutY** [reply]
Thank u for your reply. I have never had a release collection before. Will Milvus have a regular cleaning policy? Or will milvus continue to load the collection into memory until OOM finally occurs？

**Reply by yhmo** [reply]
The internal logic will try to avoid OOM. If the collection's data almost hits the memory max size, it will return an error that says the memory is not enough to load this collection.

**Reply by StormstoutY** [reply]
Alright I get it, thank you!

**Comment by zrg-team** [comment]
Hey, jump back to this topic. What happens if multiple processes load and release collection in parallel?

**Comment by xiaofan-luan** [comment]
> Hey, jump back to this topic. What happens if multiple processes load and release collection in parallel?

all ddl works in a queue. so based on the in queue time
